### ISSUE TITLE
[Feature Request] Add Popconfirm component to Element

### ISSUE THREAD

**Issue by connecteev** [ElemeFE/element#17073]
### Existing Component
No

### Component Name
PopConfirm

### Description
Add Popconfirm like https://vue.ant.design/components/popconfirm/
which looks and works great.

**Comment by Xiaoleng123** [comment]
我自己写了个，用法目前和 antd 类似，额外增添了一些属性

```vue
// popconfirm.vue
<template>
  <Popover v-bind="$attrs" v-model="sVisible" trigger="manual">
    <div class="title">
      <div class="icon">
        <slot name="icon">
          <i :class="icon" :style="`color: ${iconColor}`"></i>
        </slot>
      </div>
      <slot name="title">
        <p :class="type">{{ title }}</p>
      </slot>
    </div>
    <div class="operate-btns">
      <Button size="mini" :type="cancelType" @click="cancel">
        <slot name="cancelText">{{ cancelText }}</slot>
      </Button>
      <Button size="mini" :type="okType" @click="confirm">
        <slot name="okText">{{ okText }}</slot>
      </Button>
    </div>
    <span slot="reference" @click.stop="handleClick">
      <slot></slot>
    </span>
  </Popover>
</template>

<script>

let pre = null;

let isBinded = false;
function bindEvent () {
  if (!isBinded) {
    document.addEventListener('click', e => {
      pre && pre.setVisible(false, e);
    });
    isBinded = true;
  }
}
/**
 * 用法参考：https://vue.ant.design/components/popconfirm/
 * 额外增加check、cancelType与iconColor
 */

export default {
  name: 'pop-confirm',
  props: {
    visible: {
      type: Boolean,
      default: false
    },
    type: {
      type: String,
      default: ''
    },
    cancelText: {
      type: String,
      default: '取消'
    },
    cancelType: {
      type: String,
      default: 'default'
    },
    okText: {
      type: String,
      default: '确定'
    },
    okType: {
      type: String,
      default: 'primary'
    },
    title: {
      type: String,
      default: '你确定要执行此操作吗？'
    },
    icon: {
      type: String,
      default: 'el-icon-info'
    },
    iconColor: {
      type: String,
      default: ''
    },
    // 可以自己抛错中断，也可以返回Boolean(false以外的都认为通过)
    check: {
      type: Function,
      default () {
        return () => true;
      }
    }
  },
  data () {
    return {
      sVisible: this.visible
    };
  },
  model: {
    prop: 'visible',
    event: 'visibleChange'
  },
  watch: {
    visible (newValue) {
      this.setVisible(newValue);
    }
  },
  mounted () {
    bindEvent();
  },
  methods: {
    confirm (e) {
      this.setVisible(false, e);
      this.$emit('confirm', e);
    },
    cancel (e) {
      this.setVisible(false, e);
      this.$emit('cancel', e);
    },
    setVisible (visible, e) {
      this.sVisible = visible;
      this.$emit('visibleChange', visible, e);
    },
    handleClick (e) {
      if (pre && (pre !== this)) {
        pre.setVisible(false, e);
      }
      pre = this;

      const v = this.check();
      if (v === false) {
        return false;
      }
      if (!('visible' in this.$options.propsData)) {
        this.setVisible(!this.sVisible, e);
      }
    }
  }
};
</script>

<style lang="scss" scoped>
.operate-btns {
  text-align: right;
  margin-top: 3px;
}
.danger {
  color: red;
  font-weight: bold;
}
.title {
  .icon {
    float: left;
    font-size: 1rem;
    line-height: 1;
    margin-right: 10px;
    .el-icon-error {
      color: #fe6666;
    }

    .el-icon-info {
      color: #1890ff;
    }
  }
}
</style>

```

```vue
// 使用参考
<PopConfirm
          placement="right"
          width="245"
          type="danger"
          title="确定要删除？"
          icon="el-icon-error"
          :check="validateOperate"
          @confirm="delete"
>
```

**Comment by iamkun** [comment]
pr welcomed

**Comment by connecteev** [comment]
@Xiaoleng123 thank you
@iamkun can we add this to Element Core?

**Comment by Xiaoleng123** [comment]
> @Xiaoleng123 thank you
> @iamkun can we add this to Element Core?


The content I replied was just a "usable" component, and ElementFE will write more mature components that better fit the element design philosophy.

**Comment by connecteev** [comment]
@Xiaoleng123 makes sense. @iamkun what do you think...can this be integrated into ElementFE? Thank you.

**Comment by Xiaoleng123** [comment]
> @Xiaoleng123 makes sense. @iamkun what do you think...can this be re-purposed and integrated into ElementFE? Thank you.

@connecteev 
` iamkun added this to the 2.13.0 milestone 3 hours ago` means that it will be implemented in version 2.13.0

**Comment by connecteev** [comment]
@Xiaoleng123 oh ok! Thank you both!
